src/vs/workbench/services/editor/browser/editorService.ts

```diff
@@ -598,7 +651,159 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 		return undefined;
 	}
 
-	private findTargetGroup(input: IEditorInput, options?: IEditorOptions, group?: OpenInEditorGroup): IEditorGroup {
+	private async openEditorWith(override: EditorOverride.PICK | string, editor: IEditorInput, options: IEditorOptions | undefined, group: IEditorGroup): Promise<IEditorPane | undefined> {
+		const editorOverride = await this.findEditorOverride(override, editor, options, group);
+		if (!editorOverride) {
+			return undefined;
+		}
+
+		const [editorOverrideHandler, , targetOptions, targetGroup] = editorOverride;
+
+		return editorOverrideHandler.open(editor, targetOptions ?? options, targetGroup ?? group, OpenEditorContext.NEW_EDITOR)?.override;
+	}
+
+	private async findEditorOverride(override: EditorOverride.PICK | string, editor: IEditorInput, options: IEditorOptions | undefined, group: IEditorGroup): Promise<[IOpenEditorOverrideHandler, IOpenEditorOverrideEntry, IEditorOptions?, IEditorGroup?] | undefined> {
+
+		// We need a resource at least
+		const resource = editor.resource;
+		if (!resource) {
+			return undefined;
+		}
+
+		// Collect all overrides for resource
+		const allEditorOverrides = this.getEditorOverrides(resource, undefined, undefined);
```

Comment:
Should this call forward the `options` and `group` from the `openEditor` call or why is `undefined` explicitly passed in?

Reply:
I haven o idea, it didn't seem to change anything for me. Maybe @mjbvz knows